dvc/repo/experiments/ls.py

```diff
@@ -31,18 +31,19 @@ def ls(
             rev = [rev]
         revs = iter_revs(repo.scm, rev, num)
         rev_set = set(revs.keys())
-    ref_info_dict = exp_refs_by_baseline(repo.scm, rev_set, git_remote)
 
-    tags = repo.scm.describe(ref_info_dict.keys())
-    remained = {baseline for baseline, tag in tags.items() if tag is None}
-    base = "refs/heads"
-    ref_heads = repo.scm.describe(remained, base=base)
+    ref_info_dict = exp_refs_by_baseline(repo.scm, rev_set, git_remote)
+    baseline_names = _describe(
+        repo.scm, revs=ref_info_dict.keys(), logger=logger, remove_ref_string=False
+    )
 
     results = defaultdict(list)
     for baseline in ref_info_dict:
         name = baseline
-        if tags[baseline] or ref_heads[baseline]:
-            name = tags[baseline] or ref_heads[baseline]
+        if baseline_names[baseline]:
+            # the str() retyping is a hack to deal with the typed dictionary
+            # from _describe...probably there is a better way
+            name = str(baseline_names.get(baseline))
```

Comment:
This can probably be done in a nicer way...I haven't worked with typing much, looking forward to any suggestions.